src/Utilities/Rational.hpp

```diff
@@ -23,10 +26,15 @@ class er;
 /// about eight times as fast.
 class Rational {
  public:
-  // clang-tidy: google-explicit-constructor
-  // Treating integers as rationals is desired.
-  Rational(std::int32_t numerator = 0,  // NOLINT
-           std::int32_t denominator = 1) noexcept;
+  Rational() = default;
+  Rational(std::int32_t numerator, std::int32_t denominator) noexcept;
+
+  // Allow implicit conversion of integers to Rationals, but don't
+  // allow doubles to implicitly convert to an integer and then to a
+  // Rational.
+  template <typename T, Requires<tt::is_integer_v<T>> = nullptr>
+  // NOLINTNEXTLINE(google-explicit-constructor)
+  Rational(const T integral_value) : Rational(integral_value, 1) {}
```

Comment:
clang-tidy also seems to think this is a declaration, not a definition. Do you want to `NOLINT` that as well?

Reply:
Adding that const was literally the only change I made after running clang-tidy locally.

Reply:
Of course that's the thing that hits a clang-tidy bug :(